tests/test-server/src/main.rs

```diff
@@ -0,0 +1,252 @@
+mod handlers;
+
+use std::{net::SocketAddr, process::exit};
+
+use clap::Parser;
+
+use futures::future;
+use hickory_proto::ProtoError;
+
+use tokio::{
+    io::{AsyncReadExt, AsyncWriteExt},
+    net::{TcpListener, UdpSocket},
+    task::JoinHandle,
+};
+
+enum Transport {
+    Tcp,
+    Udp,
+}
+
+#[derive(Parser)]
+#[clap(name = "Message responder test DNS server")]
+struct Args {
+    #[clap(long = "handler")]
+    handler: Option<String>,
+    #[clap(long = "port")]
+    port: Option<u16>,
+    #[clap(long = "transport")]
+    transport: Option<String>,
+}
+
+#[tokio::main]
+async fn main() -> std::io::Result<()> {
+    let args = Args::parse();
+    let port = args.port.unwrap_or(53);
+    let transport = args.transport.unwrap_or("both".into());
+    let handler = match args.handler {
+        Some(_) => {
+            eprintln!("unknown handler");
+            exit(1);
+        }
+        None => handlers::base_handler,
```

Comment:
Is it useful to have a base handler? I'd expect we always need the invoker to specify a handler, so `Args::handler` should perhaps just be `String`?